mainwindow.cpp

```diff
@@ -1184,3 +1207,99 @@ void MainWindow::guessPasswordFinished(int exitCode, QProcess::ExitStatus exitSt
         return;
     }
 }
+
+void MainWindow::restoreSessionUI(const QString& sessionName)
+{
+    // Clear/or default optional previous session UI options that may 
+    // not be specified in the settings depending on the mode
+    foreach(QCheckBox *widget, m_ui->optionsPage->findChildren<QCheckBox*>()) {
+        widget->setChecked(false);
+    }
+    foreach(QComboBox *widget, m_ui->optionsPage->findChildren<QComboBox*>()) {
+        widget->setCurrentText("");
+    }
+    m_ui->spinBox_nbOfProcess->setMaximum(QThread::idealThreadCount());
+    m_ui->spinBox_nbOfProcess->setValue(QThread::idealThreadCount());
+    m_ui->spinBox_LimitSalts->setValue(0);
```

Comment:
These three lines of code are duplicated in the constructor too. Drop them in a separate method. You can include the previous foreach statements too and exec on a bool passed as param.